vaadin-select-flow-parent/vaadin-select-flow/src/test/java/com/vaadin/flow/component/select/data/SelectDataViewTest.java

```diff
@@ -63,10 +63,9 @@ public void setIdentifierProvider_customIdentity_itemRefreshed() {
 
         dataView.setIdentifierProvider(Item::getId);
 
-        first.setValue("changed-1");
         second.setValue("changed-2");
 
-        dataProvider.refreshItem(new Item(1L));
+        dataProvider.refreshItem(new Item(1L, "changed-1"));
```

Comment:
The previous test setup here seems flawed. Passing in a new instance should not update from the previous one. It also seems unlikely that someone would update the existing instance and then, instead of passing that same instance to `refreshItem`, pass in a new one that only matches the identity to get the corresponding item to update. So I updated the setup to cover the new change instead.